File: 927C.cpp
             // ***  ... जय श्री राम ...  ***//
             
#include<bits/stdc++.h>
using namespace std;

#define ll            long long int
#define all(x)        x.begin(),x.end()
#define for(i,a,b)    for(ll i = a ; i<b ; i++)
#define forr(j,c,d)   for(ll j = c ; j>=d ; j--)
#define yes           cout<<"yes"<<endl
#define no            cout<<"no"<<endl
#define YES           cout<<"YES"<<endl
#define NO            cout<<"NO"<<endl
#define mvi           vector<ll>
#define mvs           vector<string>
#define mpi           map<ll,ll>
#define mps           map<string,string>
#define umpi          map<string,string>
#define umps          map<ll,ll>
#define nn            cout<<endl

int main()
{
    ll t;
    cin>>t;
    while(t--){
        ll n,m;
        cin>>n>>m;
        
        mvi v(n);
        string s;

        for(i,0,n){
         cin>>v[i];   
        }

       
        cin>>s;

        ll pro = 1;
        ll allp = 1;

        for(i,0,n){

            allp = allp*v[i];
        }

       
     
       
       cout<<allp%m<<" ";

        ll j = n-1;
        ll i = 0;
        ll x = -1;
         
// cout<<allp<<" "<<endl;
     

        ll p = n;

        for(k,0,p-1){

            if(s[x+k+1]=='L') {
                ll curr = 0;

                curr = (allp/v[i])%m;

               cout<<curr<<" ";

               allp = allp/v[i];

               i++;
            }

         else if(s[x+k+1]=='R'){
                ll curr = 0;

                curr = (allp/v[j])%m;

               cout<<curr<<" ";

               allp = allp/v[j];

                j--;
            }
        }

        nn;
        nn;
        
    }
    
    return 0;
}

File: 927C.py

t = int(input())
for _ in range(t):
        n, m = map(int, input().split())
        v = list(map(int, input().split()))
        s = input()

        allp = 1
        for i in range(n):
            allp *= v[i]

        print(allp % m, end=" ")

        j = n - 1
        i = 0
        x = -1

        p = n

        for k in range(p - 1):
            if s[x + k + 1] == 'L':
                curr = (allp // v[i]) % m
                print(curr, end=" ")
                allp //= v[i]
                i += 1
            elif s[x + k + 1] == 'R':
                curr = (allp // v[j]) % m
                print(curr, end=" ")
                allp //= v[j]
                j -= 1

        print("\n\n")


